library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -442,6 +443,7 @@ public WeekView(Context context, AttributeSet attrs, int defStyleAttr) {
             mNewEventColor = a.getColor(R.styleable.WeekView_newEventColor, mNewEventColor);
             mNewEventIconDrawable = a.getDrawable(R.styleable.WeekView_newEventIconResource);
             mNewEventId = a.getInt(R.styleable.WeekView_newEventId, mNewEventId);
+            mNewEventIdentifier =  (a.getString(R.styleable.WeekView_newEventId) != null)? a.getString(R.styleable.WeekView_newEventId) : mNewEventIdentifier;
```

Comment:
I think you should also update the `mNewEventIdentifier`  when the int version is set. This can easily be solved when you use the setter `setNewEventId`. The attribute `int mNewEventId ` isn't needed anymore, you should just convert the int's to strings.